tests/ui/openid_connect_mock.rs

```diff
@@ -191,6 +233,117 @@ fn run_mock_openid_connect_server() {
+                role: Some("readonly"),
+                exc_cas: Some("ta,testbed"),
+                token_secs: Some(5),
+                refresh: true,
+                on_refresh_token_failure: Some(RefreshTokenFailureMode::InvalidGrantErrorResponse),
+                ..Default::default()
+            },
+        );  
+        known_users.insert(
+            "user-with-invalid-scope-on-refresh",
+            KnownUser {
+                role: Some("readonly"),
+                exc_cas: Some("ta,testbed"),
+                token_secs: Some(5),
+                refresh: true,
+                on_refresh_token_failure: Some(RefreshTokenFailureMode::InvalidScopeErrorResponse),
+                ..Default::default()
+            },
+        ); 
+        known_users.insert(
+            "user-with-unauthorized-client-on-refresh",
+            KnownUser {
+                role: Some("readonly"),
+                exc_cas: Some("ta,testbed"),
+                token_secs: Some(5),
+                refresh: true,
+                on_refresh_token_failure: Some(RefreshTokenFailureMode::UnauthorizedClientErrorResponse),
+                ..Default::default()
+            },
+        );
+        known_users.insert(
+            "user-with-unsupported-grant-type-on-refresh",
+            KnownUser {
+                role: Some("readonly"),
+                exc_cas: Some("ta,testbed"),
+                token_secs: Some(5),
+                refresh: true,
+                on_refresh_token_failure: Some(RefreshTokenFailureMode::UnsupportedGrantTypeErrorResponse),
+                ..Default::default()
+            },
+        );
+        known_users.insert(
+            "user-with-no-response-on-refresh",
+            KnownUser {
+                role: Some("readonly"),
+                exc_cas: Some("ta,testbed"),
+                token_secs: Some(5),
+                refresh: true,
+                on_refresh_token_failure: Some(RefreshTokenFailureMode::NoResponse),
+                ..Default::default()
+            },
+        ); 
+        known_users.insert(
+            "user-with-slow-response-on-refresh",
+            KnownUser {
+                role: Some("readonly"),
+                exc_cas: Some("ta,testbed"),
+                token_secs: Some(5),
+                refresh: true,
+                on_refresh_token_failure: Some(RefreshTokenFailureMode::SlowResponse(Duration::from_secs(25))),
+                ..Default::default()
+            },
+        ); 
+        known_users.insert(
+            "user-with-500-on-refresh",
+            KnownUser {
+                role: Some("readonly"),
+                exc_cas: Some("ta,testbed"),
+                token_secs: Some(5),
+                refresh: true,
+                on_refresh_token_failure: Some(RefreshTokenFailureMode::Error500Response),
+                ..Default::default()
+            },
+        );
+        known_users.insert(
+            "user-with-503-on-refresh",
```

Comment:
Perhaps you didn't mean to leave these as "warnings", they seem to be diagnostic in nature?

Reply:
afa3581 And eab7197. Using `dev_logger` crate to log to stdout (for now), warn, info and trace messages. Hope it's ok to pull in another dep, if not it can be incorporated into the mock server code itself.

Reply:
My initial thought is why not use the existing Fern logger dependency?

Reply:
Overlooked fern, will use that.